*************************************************************************************************
From: Karunakaran Amirthalingam
Date: Fri, 5 Dec 2025 10:49:57 +0000
Subject: [PATCH] Backward-compatibility-cmake-function-thunder-r4.patch
Reason for Change: This patch helps to resolve the generated cmakehelper function path and namespace issue 
Upstream-Status: Comcast changes
Signed-off-by: Karunakaran Amirthalingam <karunakaran_amirthalingam@cable.comcast.com>
*****************************************************************************************************
diff --git a/CMakeLists.txt b/CMakeLists.txt
index dcb447dd..f218e5a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,6 +126,17 @@ install(DIRECTORY
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${NAMESPACE}
         COMPONENT ${NAMESPACE}_Development)
 
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${NAMESPACE}.cmake"
+        RENAME "WPEFrameworkConfig.cmake"
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/WPEFramework
+        COMPONENT ${NAMESPACE}_Development)
+
+install(DIRECTORY
+            "${CMAKE_CURRENT_SOURCE_DIR}/cmake/common"
+            "${CMAKE_CURRENT_SOURCE_DIR}/cmake/templates"
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/WPEFramework
+        COMPONENT ${NAMESPACE}_Development)
+
 if(APPLE)
     # Mac needed variables
     # http://www.cmake.org/Wiki/CMake_RPATH_handling#Mac_OS_X_and_the_RPATH)
diff --git a/Source/com/com.h b/Source/com/com.h
index 8e0e8cf4..60732d05 100644
--- a/Source/com/com.h
+++ b/Source/com/com.h
@@ -38,4 +38,3 @@
 #pragma comment(lib, "com.lib")
 #endif
 
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(RPC)
diff --git a/Source/core/Portability.h b/Source/core/Portability.h
index af7f7de2..725ce0d3 100644
--- a/Source/core/Portability.h
+++ b/Source/core/Portability.h
@@ -933,18 +933,8 @@ namespace Thunder {
 }
 }
 
-// code to make (to some extend) old code which still uses the WPEFramework instead of Thunder still compile
-
-namespace WPEFramework {
-    using namespace Thunder;
-}
-
-#define WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(NESTED_NAMESPACE) \
-namespace WPEFramework { \
-namespace NESTED_NAMESPACE { \
-    using namespace Thunder::NESTED_NAMESPACE; \
-} \
-}
+// For Backward Compatibility
+#define WPEFramework Thunder
 
 extern "C" {
 
diff --git a/Source/core/core.h b/Source/core/core.h
index 0ca826b8..760d505d 100644
--- a/Source/core/core.h
+++ b/Source/core/core.h
@@ -115,5 +115,4 @@
 #pragma comment(lib, "core.lib")
 #endif
 
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(Core)
 #endif
diff --git a/Source/cryptalgo/cryptalgo.h b/Source/cryptalgo/cryptalgo.h
index 004bd15e..fcb13ad4 100644
--- a/Source/cryptalgo/cryptalgo.h
+++ b/Source/cryptalgo/cryptalgo.h
@@ -37,4 +37,3 @@
 #pragma comment(lib, "cryptalgo.lib")
 #endif
 
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(Crypto)
diff --git a/Source/extensions/processcontainers/processcontainers.h b/Source/extensions/processcontainers/processcontainers.h
index f862bee4..f30804b2 100644
--- a/Source/extensions/processcontainers/processcontainers.h
+++ b/Source/extensions/processcontainers/processcontainers.h
@@ -26,4 +26,3 @@
 #include "IProcessContainers.h"
 #include "ContainerAdministrator.h"
 
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(ProcessContainers)
diff --git a/Source/extensions/warningreporting/warningreporting.h b/Source/extensions/warningreporting/warningreporting.h
index 1e46cbb8..e1421364 100644
--- a/Source/extensions/warningreporting/warningreporting.h
+++ b/Source/extensions/warningreporting/warningreporting.h
@@ -30,4 +30,3 @@
 #pragma comment(lib, "warningreporting.lib")
 #endif
 
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(WarningReporting)
diff --git a/Source/messaging/messaging.h b/Source/messaging/messaging.h
index 128e2cce..0b1ea8d6 100644
--- a/Source/messaging/messaging.h
+++ b/Source/messaging/messaging.h
@@ -38,4 +38,3 @@
 #pragma comment(lib, "messaging.lib")
 #endif
 
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(Messaging)
diff --git a/Source/plugins/plugins.h b/Source/plugins/plugins.h
index 70f6ac35..2f5956c8 100644
--- a/Source/plugins/plugins.h
+++ b/Source/plugins/plugins.h
@@ -52,5 +52,3 @@
 #pragma comment(lib, "plugins.lib")
 #endif
 
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(Plugin)
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(PluginHost)
diff --git a/Source/websocket/websocket.h b/Source/websocket/websocket.h
index 035822ad..c83d72f2 100644
--- a/Source/websocket/websocket.h
+++ b/Source/websocket/websocket.h
@@ -38,5 +38,3 @@
 #pragma comment(lib, "websocket.lib")
 #endif
 
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(Web)
-WPEFRAMEWORK_NESTEDNAMESPACE_COMPATIBILIY(JSONRPC)
